src/sparseml/core/logger/logger.py

```diff
@@ -1123,6 +1124,25 @@ def log_string(
                     level=level,
                 )
 
+    @contextmanager
+    def time(self, tag: Optional[str] = None, *args, **kwargs):
+        """
+        Context manager to log the time it takes to run the block of code
+
+        Usage:
+        >>> with LoggerManager().system.time("my_block"):
+        >>>    time.sleep(1)
+
+        :param tag: identifying tag to log the values with
+        """
+
+        start = time.time()
+        yield
+        elapsed = time.time() - start
+        if not tag:
+            tag = f"{DEFAULT_TAG}_time"
+        self.log_string(tag=tag, string=f"{elapsed:.3f}s", *args, **kwargs)
```

Comment:
We should be logging a float not a string, will allow for easier plotting and such